pkg/openstack/test.go

```diff
@@ -0,0 +1,60 @@
+package openstack
+
+import (
+	"context"
+
+	"github.com/openstack-k8s-operators/lib-common/modules/common/condition"
+	"github.com/openstack-k8s-operators/lib-common/modules/common/configmap"
+	"github.com/openstack-k8s-operators/lib-common/modules/common/helper"
+	"github.com/openstack-k8s-operators/lib-common/modules/common/util"
+
+	corev1beta1 "github.com/openstack-k8s-operators/openstack-operator/apis/core/v1beta1"
+	ctrl "sigs.k8s.io/controller-runtime"
+)
+
+const (
+	// config map generated by Keyston operator containing clouds.yaml
+	CloudsConfigMap       = "openstack-config"
+	TestOperatorConfigMap = "test-operator-config"
+
+	TempestImageKey = "tempest-image"
+	TobikoImageKey  = "tobiko-image"
+	CloudsKey       = "clouds-secret-name"
+)
+
+func ReconcileTest(ctx context.Context, instance *corev1beta1.OpenStackControlPlane, version *corev1beta1.OpenStackVersion, helper *helper.Helper) (ctrl.Result, error) {
+	customData := map[string]string{
+		TempestImageKey: *version.Status.ContainerImages.TestTempestImage,
+		TobikoImageKey:  *version.Status.ContainerImages.TestTobikoImage,
```

Comment:
I think this is potentially unsafe if in downstream we won't have the tobiko variable set (there is no downstream build).